pydm/data_plugins/local_plugin.py

```diff
@@ -1,332 +1,427 @@
+                self._type_kwargs['dtype'] = np.dtype(dtype)
+                return self._type_kwargs
+            except ValueError:
+                logger.debug('Cannot convert dtype')
+
+        return self._type_kwargs
+
+    def send_access_state(self):
+        """
+        Send True for all the widgets using Local Plugin.
+
+        Returns
+        -------
+        None.
+
+        """
+        self.write_access_signal.emit(True)
 
-        def send_connection_state(self, conn=None):
-            """
-            Send an update on our connection state to every listener.
+    def convert_value(self, value, value_type):
+        """
+        Convert values to their appropriate types.
+
+        Parameters
+        ----------
+        value :
+            Data for this variable.
+        value_type : str
+            Data type intended for this variable.
 
-            :param conn: True if attribute exists, False otherwise.
-            :type conn:  bool
-            """
-            self.connection_state_signal.emit(conn)
+        Returns
+        -------
+            The data for this variable converted to its appropriate type
 
-        def is_connected(self):
-            """
-            Return True if we can get a value.
-            """
+        """
+
+        _type = self._data_types.get(value_type)
+        if _type is not None:
             try:
-                self.get_value()
-                return True
-            except:
-                return False
-
-        @Slot(int)
-        @Slot(float)
-        @Slot(str)
-        @Slot(np.ndarray)
-        def put_value(self, value):
-            """
-            Set our object attribute's value. If the attribute is a function,
-            we will execute it.
-
-            :param value: The value we'd like to set in our object.
-            :type value:  int, float, str, or np.ndarray.
-            """
-            # Field: replace value
-            if self.nargs is None:
-                try:
-                    setattr(obj, self.attr, value)
-                except:
-                    return
-            # Function of zero arguments: call function
-            elif self.nargs == 0:
-                try:
-                    getattr(obj, self.attr)()
-                except:
-                    return
-            # Function of one argument: call function with value
-            elif self.nargs == 1:
-                try:
-                    getattr(obj, self.attr)(value)
-                except:
-                    return
-            # Function with many arguments: distribute arguments to args
-            elif self.nargs > 1:
-                try:
-                    getattr(obj, self.attr)(*value)
-                except:
-                    return
+                if value_type == "array":
+                    value = value.replace("[", "").replace("]", "").replace(",", " ")
+                    value = shlex.split(value)
```

Comment:
I think `ast.literal_eval` would be preferable to this. 
```suggestion
                    value = ast.literal_eval(value)
```